src/averageAges.js

```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter((p) => p.sex === 'f'
+      && people.find((p1) => p1.mother === p.name))
+    : people.filter((p) => p.sex === 'f');
```

Comment:
it's better to set full variable name: person, supposedDaughter
```suggestion
  const women = withChildren
    ? people.filter((person) => person.sex === 'f'
      && people.find((supposedDaughter) => supposedDaughter.mother === person.name))
    : people.filter((person) => person.sex === 'f');
```